netbox/dcim/api/views.py

```diff
@@ -343,6 +343,21 @@ class DeviceViewSet(CustomFieldModelViewSet):
     )
     filterset_class = filters.DeviceFilterSet
 
+    def get_queryset(self):
```

Comment:
Can we move this to a mixin to avoid replicating the code in `virtualization/api/views.py`?